test/lib/failure_group_api_test.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class FailureGroupApiTest < ActiveSupport::TestCase
+  it 'should get the failure groups' do
+    VCR.use_cassette('failure_groups', match_requests_on: [:path]) do
+      api = FailureGroupApi.new.fetch
+      assert JSON.parse(api)['description'] = 'No branch found'
+    end
+  end
+
+  it 'should get the failure groups for given id' do
+    VCR.use_cassette('failure_groups') do
```

Comment:
Name the VCR cassette by the corresponding controller + action name. In this case `failure_groups_find_by_job`.